tracer/src/Datadog.Trace/Ci/IntelligentTestRunnerClient.cs

```diff
@@ -71,54 +76,58 @@ public IntelligentTestRunnerClient(string workingDirectory, CIVisibilitySettings
+                _settingsUrl = new UriBuilder(agentlessUrl) { Path = settingsUrlPath }.Uri;
+                _searchCommitsUrl = new UriBuilder(agentlessUrl) { Path = searchCommitsUrlPath }.Uri;
+                _packFileUrl = new UriBuilder(agentlessUrl) { Path = packFileUrlPath }.Uri;
+                _skippableTestsUrl = new UriBuilder(agentlessUrl) { Path = skippableTestsUrlPath }.Uri;
+            }
+            else
             {
-                Path = "api/v2/ci/tests/skippable"
-            }.Uri;
+                _settingsUrl = new UriBuilder(
+                    scheme: "https",
+                    host: "api." + _settings.Site,
+                    port: 443,
+                    pathValue: settingsUrlPath).Uri;
+
+                _searchCommitsUrl = new UriBuilder(
+                    scheme: "https",
+                    host: "api." + _settings.Site,
+                    port: 443,
+                    pathValue: searchCommitsUrlPath).Uri;
+
+                _packFileUrl = new UriBuilder(
+                    scheme: "https",
+                    host: "api." + _settings.Site,
+                    port: 443,
+                    pathValue: packFileUrlPath).Uri;
+
+                _skippableTestsUrl = new UriBuilder(
+                    scheme: "https",
+                    host: "api." + _settings.Site,
+                    port: 443,
+                    pathValue: skippableTestsUrlPath).Uri;
+            }
         }
         else
         {
-            _settingsUrl = new UriBuilder(
-                scheme: "https",
-                host: "api." + _settings.Site,
-                port: 443,
-                pathValue: "api/v2/libraries/tests/services/setting").Uri;
-
-            _searchCommitsUrl = new UriBuilder(
-                scheme: "https",
-                host: "api." + _settings.Site,
-                port: 443,
-                pathValue: "api/v2/git/repository/search_commits").Uri;
-
-            _packFileUrl = new UriBuilder(
-                scheme: "https",
-                host: "api." + _settings.Site,
-                port: 443,
-                pathValue: "api/v2/git/repository/packfile").Uri;
-
-            _skippableTestsUrl = new UriBuilder(
-                scheme: "https",
-                host: "api." + _settings.Site,
-                port: 443,
-                pathValue: "api/v2/ci/tests/skippable").Uri;
+            // Use Agent EVP Proxy
+            _useEvpProxy = true;
+            var agentUrl = _settings.TracerSettings.Exporter.AgentUri;
+            _settingsUrl = new UriBuilder(agentUrl) { Path = $"/evp_proxy/v1/{settingsUrlPath}" }.Uri;
+            _searchCommitsUrl = new UriBuilder(agentUrl) { Path = $"/evp_proxy/v1/{searchCommitsUrlPath}" }.Uri;
+            _packFileUrl = new UriBuilder(agentUrl) { Path = $"/evp_proxy/v1/{packFileUrlPath}" }.Uri;
+            _skippableTestsUrl = new UriBuilder(agentUrl) { Path = $"/evp_proxy/v1/{skippableTestsUrlPath}" }.Uri;
```

Comment:
Word of warning, if `agentUrl` has a path, this will drop it. Unlikely, just flagging it. Do we also need to think about UDS/Named pipes here? 🤔

Reply:
true, but the AgentUrl is from the APM agent configuration, and can be set directly or through AGENT_HOST + AGENT_PORT. So, we don't expect a path here. Also we are matching the path for the agent endpoint (as /info shows).

Reply:
UDS or Named pipes support can come later (if we have that request from customers), but not for now.

Reply:
> AGENT_HOST + AGENT_PORT. So, we don't expect a path here. 

I would be careful about this assumption, K8s in particular I've seen extra paths creeping in for simplicity with ingresses etc. But I agree, it's _unlikely_ to be an issue, just something to consider, especially as you could use `UriHelpers.Combine(agentUrl, "/my/path");`

> UDS or Named pipes support can come later (if we have that request from customers), but not for now

👍